### PR TITLE
Fix debugger bugs

### PR DESCRIPTION
**Pull request by Kuinox**
Fixed a xplat bug that affect unix&windows, but most of the times unix.  
Reworked the pipes swaps logics so it works on linux, we were leaking a pipe in our process which was the reason it never closed.

### CODE REVIEW

File: src/Draco.Debugger.Tests/IoUtilsTests.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+using System.Diagnostics;
+using Draco.Debugger.IO;
+using Xunit;
+
+namespace Draco.Debugger.Tests;
+public class IoUtilsTests
```
Comment (LPeter1997) [inline src/Draco.Debugger.Tests/IoUtilsTests.cs:6]:
```suggestion
namespace Draco.Debugger.Tests;

public sealed class IoUtilsTests
```

File: src/Draco.Debugger/Platform/UnixPlatformMethods.cs [hunk 2]
```diff
@@ -24,23 +24,23 @@ private enum FileDescriptor : int
...
-        if (newCopy == -1) throw new InvalidOperationException($"could not set {old} handle");
-
-        return oldCopy;
+        // Duplicate the existing file descriptors to get their handles
+        var stdin = (IntPtr)dup((int)FileDescriptor.STDIN_FILENO);
+        var stdout = (IntPtr)dup((int)FileDescriptor.STDOUT_FILENO);
+        var stderr = (IntPtr)dup((int)FileDescriptor.STDERR_FILENO);
+        //var stdin = (IntPtr)FileDescriptor.STDIN_FILENO; // investigate why this doesn't works.
+        //var stdout = (IntPtr)FileDescriptor.STDOUT_FILENO;
+        //var stderr = (IntPtr)FileDescriptor.STDERR_FILENO;
```
Comment (LPeter1997) [inline src/Draco.Debugger/Platform/UnixPlatformMethods.cs:35]: Let's not leave in commented code

File: src/Draco.Debugger/Platform/IPlatformMethods.cs [hunk 3]
```diff
@@ -8,9 +8,13 @@ namespace Draco.Debugger.Platform;
...
     /// <summary>
-    /// Replaces the standard IO handles for this process.
+    /// Gets the standard IO handles of the current process.
     /// </summary>
-    /// <param name="newHandles">The new handles to be used by the process.</param>
-    /// <returns>The old IO handles.</returns>
-    public IoHandles ReplaceStdioHandles(IoHandles newHandles);
+    /// <returns></returns>
+    public IoHandles GetStdioHandles();
+    /// <summary>
```
Comment (LPeter1997) [inline src/Draco.Debugger/Platform/IPlatformMethods.cs:15]:
```suggestion
    /// <returns></returns>
    public IoHandles GetStdioHandles();
    
    /// <summary>
```

File: src/Draco.Debugger/Platform/IPlatformMethods.cs [hunk 4]
```diff
@@ -8,9 +8,13 @@ namespace Draco.Debugger.Platform;
...
 {
     /// <summary>
-    /// Replaces the standard IO handles for this process.
+    /// Gets the standard IO handles of the current process.
     /// </summary>
-    /// <param name="newHandles">The new handles to be used by the process.</param>
-    /// <returns>The old IO handles.</returns>
-    public IoHandles ReplaceStdioHandles(IoHandles newHandles);
+    /// <returns></returns>
+    public IoHandles GetStdioHandles();
```
Comment (LPeter1997) [inline src/Draco.Debugger/Platform/IPlatformMethods.cs:14]: Get is not a nice name, more like Fork or Copy?

File: src/Draco.Debugger/Platform/UnixPlatformMethods.cs (hunk 2 above)
Comment (Kuinox) [reply src/Draco.Debugger/Platform/UnixPlatformMethods.cs:35]: Oops

File: src/Draco.Debugger/Platform/IPlatformMethods.cs (hunk 4 above)
Comment (Kuinox) [reply src/Draco.Debugger/Platform/IPlatformMethods.cs:14]: The behavior is platform dependent, it doesn't copy on Windows.

**Review by LPeter1997 — APPROVED** [review]

**Pull request merged**